benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -413,12 +413,12 @@ parseMany n =
     . fmap Sum
 
 {-# INLINE parseIterate #-}
-parseIterate :: MonadThrow m => Int -> Stream m Int -> m ()
+parseIterate :: MonadFail m => Int -> Stream m Int -> m ()
 parseIterate n =
       Stream.fold Fold.drain
     . fmap getSum
-    . Stream.parseIterate (PR.fromFold . Fold.take n . Fold.sconcat) (Sum 0)
-    . fmap Sum
+    . Stream.rights.Stream.parseIterate (PR.fromFold . Fold.take n . Fold.sconcat) (Right $ Sum 0)
```

Comment:
Always use spaces around any operator. `Stream.rights . Stream.parseIterate`. Also if you notice each `.` starts from a new line.